### ISSUE TITLE
How to set -Pgit.root=path of .git for nebula.rlease plugin to work on Google cloudbuild

### ISSUE THREAD

**Issue by vmahuvak** [nebula-plugins/nebula-release-plugin#156]
* What went wrong:
Step #0: Task 'build -Pgit.root=/workspace/mytest --scan --debug' not found in root project 'mytest'.

**Comment by vmahuvak** [comment]
I have set git.root=-P<path .git for may project> as it is in a directory higher level than my gradle project. It works file when I run on my Mac

gradle final -Dorg.ajoberstar.grgit.auth.username= -Dorg.ajoberstar.grgit.auth.password= -Pgit.root=/Users/vikrammahuvakar/projects/mytest final

but it fails on Cloudbuild as I cannot set -Pgit.root=<..>
I tried
-Pgit.root=$REPO_NAME
-Pgit.root=/workspace

Any suggestion how I can set this for Cloudbuild in my cloudbuild.yaml file??

**Comment by chali** [comment]
Hi @vmahuvak, unfortunately, we don't have much experience with Cloudbuild. But from the error, it looks like the whole command line is considered to be a single task name. I would recommend searching in that direction. In your case task is just `build`. This looks like an example which could help https://github.com/GoogleCloudPlatform/cloud-builders/tree/master/gradle

**Comment by vmahuvak** [comment]
Thanks for the information. It works on my local Mac and Windows 10 machines. Just with Cloudbuild I was not sure if the .git is cloned. I just had one more question. How the nebula.release groovy get's version.previousVersion parameter value. Does it get from FETCH_HEAD file in.git directory of from .git/ref/tag directory with has one file which keeps previous tag version in it.
Please let me know 

Thanks,
Vikram

    On Monday, July 29, 2019, 4:41:50 PM PDT, Martin Chalupa <notifications@github.com> wrote:  
 
 
Hi @vmahuvak, unfortunately, we don't have much experience with Cloudbuild. But from the error, it looks like the whole command line is considered to be a single task name. I would recommend searching in that direction. In your case task is just build. This looks like an example which could help https://github.com/GoogleCloudPlatform/cloud-builders/tree/master/gradle

**Comment by chali** [comment]
I'm not sure what exactly is this `version.previousVersion` parameter but in the plugin, we are using `jgit` library for all the git operations. I would guess that eventually, it reads from `.git` directory but we don't do that directly.

**Comment by rpalcolea** [comment]
Closing this due to lack of activity. Please let us know how things went for you

**Comment by vmahuvak** [comment]
Hi 
All I was asking about adding a new property version.previousVersion=<e.g. 1.0.x>which can be used by nebula-release to increment version number. Google Cloudbuild doesn't get all the information in .git in a /workspace folder. I have resolved this by using release.version=<x.x.x>. I set this every time which helps nebula-release to use that as tag release version. This and some other setup on cloudbuid.yaml I have resolved all the issues including git authentication password/token passing in cloudbuild.yaml

Thanks,

Vikram
